### PR TITLE
[LM-10924] API Key Security

### PR DESCRIPTION
**Pull request by devansh-mohata**
## Description

**Features**
- Introduced API Key Security

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fix for breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: lm-feed-sample-app/lm-feed-sample-app/FeedTokenHandler.swift [hunk 1]
```diff
@@ -0,0 +1,112 @@
+//
+//  FeedTokenHandler.swift
```
Comment (devansh-mohata) [inline lm-feed-sample-app/lm-feed-sample-app/FeedTokenHandler.swift:2]: Remove file

File: lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift [hunk 2]
```diff
@@ -92,43 +144,74 @@ public class LMFeedCore {
...
     public func didReceiveNotification(_ notification: UNNotificationRequest, completion: ((Result<LMViewController, LMFeedError>) -> Void)?) {
         guard Self.isInitialized,
-              let apiKey = LocalPreferences.apiKey,
-              let userUUID = LocalPreferences.userObj?.clientUUID,
-              let userName = LocalPreferences.userObj?.name,
               let route = notification.content.userInfo["route"] as? String else {
             completion?(.failure(.feedNotInitialized))
             return
         }
+        // TODO: Ask Ishaan
```
Comment (devansh-mohata) [inline lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift:151]: Check

File: lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift [hunk 3]
```diff
@@ -92,43 +144,74 @@ public class LMFeedCore {
...
+//                }
+//            }
+//        }
+    }
+}
+
+
+extension LMFeedCore: LMFeedSDKCallback {
+    public func onAccessTokenExpiredAndRefreshed(accessToken: String, refreshToken: String) {
+        print("\(#function)-\(#file)")
```
Comment (devansh-mohata) [inline lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift:188]: remove print statement

File: lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift [hunk 4]
```diff
@@ -92,43 +144,74 @@ public class LMFeedCore {
...
+        coreCallback?.onAccessTokenExpiredAndRefreshed(accessToken: accessToken, refreshToken: refreshToken)
+    }
+    
+    public func onRefreshTokenExpired(_ completionHandler: (((accessToken: String, refreshToken: String)?) -> Void)?) {
+        print("\(#function)-\(#file)")
```
Comment (devansh-mohata) [inline lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift:193]: remove print statement

File: lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/Screens/CreatePost/LMFeedCreatePostScreen.swift [hunk 5]
```diff
@@ -7,6 +7,7 @@
 
 import AVKit
 import BSImagePicker
+import LikeMindsFeed
```
Comment (devansh-mohata) [inline lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/Screens/CreatePost/LMFeedCreatePostScreen.swift:10]: check import

**Review by pushpendralike — APPROVED** [review]

**Pull request merged**